### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by joyqvq**
updated it to newly included crates and functions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,45 @@
-# fastcrypto
-Common cryptographic library used in software at Mysten Labs.
+# [fastcrypto]
+
+[![crate][crate-image]][crate-link]
+[![Docs][docs-image]][docs-link]
+![Apache2/MIT licensed][license-image]
+![Rust Version][rustc-image]
+
+fastcrypto` is a common cryptography library used by software at Mysten Labs. It is a wrapper library around several carefully selected crates with following considerations:
```
Comment (kchalkias) [inline README.md:8]: Please add "but on purpose published independently encouraging reusability across different applications and domains"

File: README.md [hunk 2]
```diff
@@ -1,25 +1,45 @@
...
+![Apache2/MIT licensed][license-image]
+![Rust Version][rustc-image]
+
+fastcrypto` is a common cryptography library used by software at Mysten Labs. It is a wrapper library around several carefully selected crates with following considerations: 
+- Security: Whether the libraries are vulnerable to known attack vectors or possible misuses. 
+- Performance: Whether the crate performs cryptographic operations with speed after extensive benchmarking. This is critical for the Sui Network to be performant when signing and verifying large amounts of transactions and certificates. 
+- Determinism: Whether the signature is non-malleable.
+- Popularity: Whether the library is used by other consensus critical systems. 
+
+Furthermore, we extends the selected libraries with additional features:
```
Comment (kchalkias) [inline README.md:14]: extend

File: README.md [hunk 3]
```diff
@@ -1,25 +1,45 @@
...
+    - Ed25519: Backed by [`ed25519-consensus`](https://github.com/penumbra-zone/ed25519-consensus) crate. Compliant to [ZIP-215](https://zips.z.cash/zip-0215) that defines the signature validity that is lacking from RFC8032 but critical for consensus algorithms. `ed25519-dalek` is fully deprecated due to the recently discovered [Chalkias double pub-key api vulnerability](https://github.com/MystenLabs/ed25519-unsafe-libs).
```
Comment (kchalkias) [inline README.md:34]: lets mention ed25519 before secp256k1

File: README.md [hunk 4]
```diff
@@ -1,25 +1,45 @@
...
+- Security: Whether the libraries are vulnerable to known attack vectors or possible misuses. 
+- Performance: Whether the crate performs cryptographic operations with speed after extensive benchmarking. This is critical for the Sui Network to be performant when signing and verifying large amounts of transactions and certificates. 
+- Determinism: Whether the signature is non-malleable.
+- Popularity: Whether the library is used by other consensus critical systems. 
+
+Furthermore, we extends the selected libraries with additional features:
+- Robust testing framework: [Wycheproof tests](https://github.com/google/wycheproof) and [prop tests](https://altsysrq.github.io/proptest-book/intro.html) are added when possible to protect against arbitrary inputs and crafted edge cases.
+- Zeroization: Sensitive private key materials are cleared from memory securely and proactively when it goes out of scope using [zeroize](https://docs.rs/zeroize/latest/zeroize/) trait.
+- Serialization: Effective and standardized serialization are required.
+
```
Comment (kchalkias) [inline README.md:18]: This library will be continuously updated with more schemes and faster / more secure implementations based on benchmarking results, RFC updates and audit inputs.

File: README.md [hunk 5]
```diff
@@ -1,25 +1,45 @@
...
+    - Ed25519: Backed by [`ed25519-consensus`](https://github.com/penumbra-zone/ed25519-consensus) crate. Compliant to [ZIP-215](https://zips.z.cash/zip-0215) that defines the signature validity that is lacking from RFC8032 but critical for consensus algorithms. `ed25519-dalek` is fully deprecated due to the recently discovered [Chalkias double pub-key api vulnerability](https://github.com/MystenLabs/ed25519-unsafe-libs).
+    - BLS12-381: Backed by [`blst`](https://github.com/supranational/blst) crate written in Assembly and C that optimizes for performance and security. G1 and G2 points are serialized following [ZCash specification](https://github.com/supranational/blst#serialization-format) in compressed format. Provides methods for verifying signatures in the G1 group against public keys in the G2 group. Provides methods for aggregating signatures and fast verifying aggregated signatures, where public keys are assumed to be verified for proof of possession.
+
+- Utility Functions that serves as the underlying rust implementation for the Move smart contract API.
```
Comment (kchalkias) [inline README.md:37]: use RUST and api

File: README.md [hunk 6]
```diff
@@ -1,25 +1,45 @@
...
+    - BLS12-381: Backed by [`blst`](https://github.com/supranational/blst) crate written in Assembly and C that optimizes for performance and security. G1 and G2 points are serialized following [ZCash specification](https://github.com/supranational/blst#serialization-format) in compressed format. Provides methods for verifying signatures in the G1 group against public keys in the G2 group. Provides methods for aggregating signatures and fast verifying aggregated signatures, where public keys are assumed to be verified for proof of possession.
+
+- Utility Functions that serves as the underlying rust implementation for the Move smart contract API. 
+    - HKDF: An HMAC-based key derivation function based on [RFC-5869](https://tools.ietf.org/html/rfc5869), to derive keypairs with a salt and an optional domain for the given keypair. This requires choosing an HMAC function that expands precisely to the byte length of a Private key for the chosen KeyPair parameter.
```
Comment (kchalkias) [inline README.md:38]: private key

**Review by kchalkias — APPROVED** [review]
Awesome!
